### PR TITLE
Fault tolerance feature and UC on the queue table

### PR DESCRIPTION
**Pull request by natabeck**

### CODE REVIEW

File: engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java [hunk 1]
```diff
@@ -47,70 +41,107 @@
...
-        final AtomicInteger messagesCounter = new AtomicInteger(0);
-
-        dispatch(messages, ExecutionMessage.class, new Handler<ExecutionMessage>() {
-            @Override
-            public void handle(List<ExecutionMessage> messages) {
-                messagesCounter.addAndGet(messages.size());
-                queueDispatcher.dispatch(messages);
+    private void dispatchBulk(List<? extends Serializable> messages, String bulkNumber, String workerUuid){
+        try {
+            for(int i=0; i<3; i++){
```
Comment (orius123) [inline engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java]: please expose constants

File: engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java [hunk 2]
```diff
@@ -47,70 +41,107 @@
...
+                try{
+                    //Dispatch list of messages in NEW transaction!!!
+                    dispatcherHelperService.dispatchBulk(messages, bulkNumber, workerUuid);
+
+                    break;  //If the insert of bulk worked - stop retries
+                }
+                catch (Exception bulkException){
+                    //If it is the third loop and we still got exception - throw exception
+                    if(i == 2){
```
Comment (orius123) [inline engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java]: please constants

**Comment by orius123** [comment]
@natabeck this is a pretty big change to solve a small issue.
can you please explain the motivation? what are the changes in English?
are all of this changes related to the issue with draining the outbuffer in a bulk?

File: engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java (hunk 1 above)
Comment (natabeck) [inline engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java]: Done

File: engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java [hunk 3]
```diff
@@ -47,70 +44,107 @@
...
+                    //Dispatch list of messages in NEW transaction!!!
+                    dispatcherHelperService.dispatchBulk(messages, bulkNumber, workerUuid);
+
+                    break;  //If the insert of bulk worked - stop retries
+                }
+                catch (Exception bulkException){
+                    //If it is the third loop and we still got exception - throw exception
+                    if(i == (dispatchBulkRetries - 1) ){
+                        logger.error("Failed to dispatch bulk of messages to the queue for 3 times, going to check DB connection! ", bulkException);
+                        throw bulkException;
```
Comment (orius123) [inline engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java]: dispatchBulkRetries  - also in the message

Comment (natabeck) [inline engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/OrchestratorDispatcherServiceImpl.java]: done

**Comment by orius123** [comment]
@natabeck can we close this PR?

**Comment by natabeck** [comment]
Yes

**Pull request closed without merging**